src/main/java/app/linguistai/bmvp/service/wordbank/UnknownWordService.java

```diff
@@ -343,6 +344,45 @@ public ROwnerUnknownWordList deleteList(UUID listId, String email) throws Except
         }
     }
 
+    @Override
+    @Transactional
+    public UnknownWordList getRandomActiveUnknownWordList(UUID userId) throws Exception {
+        try {
+            List<UnknownWordList> activeLists = listRepository.findByUserId(userId)
+                .stream()
+                .filter(UnknownWordList::getIsActive)
+                .toList();
+
+            if (activeLists.isEmpty()) {
+                throw new NotFoundException("No active unknown word lists found for user with ID: [" + userId + "].");
+            }
+
+            return activeLists.get(new Random().nextInt(activeLists.size()));
+        }
+        catch (Exception e) {
+            System.out.println("ERROR: Could not retrieve random active unknown word list.");
+            throw e;
+        }
+    }
+
+    @Override
+    @Transactional
+    public String getRandomWordFromList(UUID listId) throws Exception {
+        try {
+            List<UnknownWord> words = wordRepository.findByOwnerListListId(listId);
+
+            if (words.isEmpty()) {
+                throw new NotFoundException("No words found in the unknown word list with ID: [" + listId + "].");
+            }
```

Comment:
A probable 404 issue here, similar to getRandomActiveUnknownWordList

Reply:
Tested with a new user and confirmed a single active list exists with no words, login is successful. However 400 (bad request) returns when a new UseWord quest is attempted to be given to the user.

Since the user must have at lease one active word list (with at lease one -number is not discussed as far as I remember- unknown word) at one time (rule), this should give an error in my opinion.